lucene/core/src/java/org/apache/lucene/codecs/lucene80/Lucene80DocValuesConsumer.java

```diff
@@ -736,49 +736,92 @@ private void doAddSortedField(FieldInfo field, DocValuesProducer valuesProducer)
   private void addTermsDict(SortedSetDocValues values) throws IOException {
     final long size = values.getValueCount();
     meta.writeVLong(size);
-    meta.writeInt(Lucene80DocValuesFormat.TERMS_DICT_BLOCK_SHIFT);
+    boolean compress =
+        Lucene80DocValuesFormat.Mode.BEST_COMPRESSION == mode
```

Comment:
Is it possible to just enable compression for `SortedDocValues` but not `BinaryDocValues`? Since compressed `BinaryDocValues` is proved to be harmful for our service but we still want to try this one out...

Reply:
I didn't find a good solution to resolve this problem, since terms dict compression could not be enabled by default currently.....there will be two options:
1.  introduce more configuration/mode.  it will make things more complicated.
2. optimize BinaryDocValues compression.

We will try this feature in our cluster first, and see the actual performance..so can we resolve this in another issue? or introduce more configurations in this patch?

Reply:
@zhaih could you be more specific about this? Could you post something in the Jira issue discussion to explain how this compression is harmful for BinaryDocValues? It needs to be more visible than just in this review.

Reply:
Terms dict compression and BinaryDocValues compression now share the same configuration...it means these two compression will be enabled and disabled together...But BinaryDocValues compression will cause performance reduction.. Please check the discussions from [LUCENE-9378](https://issues.apache.org/jira/browse/LUCENE-9378) :)

Reply:
Sorry for late response. I agree we could solve it in a follow-up issue. And I could still test this via a customized PerFieldDocValuesFormat, thank you!